marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -185,6 +185,7 @@ public void awaitCompletion(long timeout, TimeUnit unit) throws InterruptedExcep
     public Job run() {
         runningThread = null;
         Job job = createJob();
+        jobUpdate = new JobUpdate(hubConfig.newJobDbClient());
```

Comment:
I was wondering any benefit to create another thread for step runner (write/read) since we has spawned a thread in flowrunner and we execute step tasks concurrently...

Reply:
I am not sure if I understand this correctly.  StepRunners are just adoption of the previous 4.x FlowRunnerImpl